### PR TITLE
Reintroduce Clang Sanitizer

### PR DESCRIPTION
**Pull request by franzpoeschel**
Re-add ASAN and UBSAN tests to CI.

### CODE REVIEW

**Comment by ax3l** [comment]
Since the tests are relatively long-running, let's keep the clang-tidy & sanitizer matrix entries separate

**Comment by franzpoeschel** [comment]
Note: I noticed that the `clang10_py38_nompi_h5_ad1_ad2_libcpp` test does not build any ADIOS backends
EDIT: The added sanitizer test now does exactly what that test promises to do, but additionally with sanitization. So, I'd say we can just keep that one as a no-Adios-test and rename it accordingly?

File: .github/ci/spack-envs/clang10_nopy_ompi_h5_ad1_ad2_/spack.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# This is a Spack environment file.
```
Comment (ax3l) [inline .github/ci/spack-envs/clang10_nopy_ompi_h5_ad1_ad2_/spack.yaml]: It looks like the directory name here ends with an `_`, which looks like a typo

File: .github/workflows/unix.yml [hunk 2]
```diff
@@ -20,14 +20,44 @@ jobs:
       run: |
         eval $(spack env activate --sh .github/ci/spack-envs/clangtidy_nopy_ompi_h5_ad1_ad2/)
         spack install
```
Comment (ax3l) [inline .github/workflows/unix.yml]: Just reverting the white space change:
```suggestion
        spack install

```

**Review by ax3l — APPROVED** [review]
Wuhu, thanks for re-introducing this!

Safety first! :)

**Comment by ax3l** [comment]
Wow, this still runs for an hour. But no worries, we will think about optimizing this if we need to.

**Pull request merged**